app/src/main/java/ch/epfl/sweng/SDP/utils/network/ConnectivityWrapper.java

```diff
@@ -0,0 +1,102 @@
+package ch.epfl.sweng.SDP.utils.network;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.support.annotation.VisibleForTesting;
+
+import ch.epfl.sweng.SDP.firebase.Database;
+import ch.epfl.sweng.SDP.game.WaitingPageActivity;
+
+/**
+ * Connectivity wrapper that registers the the network receiver and unregisters it.
+ */
+public final class ConnectivityWrapper {
+
+    private static final String TOP_ROOM_NODE_ID = "realRooms";
+    private static NetworkStateReceiver networkStateReceiver = null;
+
+    private ConnectivityWrapper() {}
```

Comment:
Inside the constructor you should check for instance != null, and if it’s the case, an IllegalStateException should be thrown